File: R/CfromR/myRLib.R
dyn.load("myRLib.so")

mySum <- function(a) {
    n <- length(a);
    result <- .C("mySum", as.double(a), as.integer(n), s = double(1));
    result$s;
}

myMult <- function(a, lambda) {
    n <- length(a);
    result <- .C("myMult", as.double(a), as.integer(n),
                 as.double(lambda), m = double(n));
    result$m;
}

File: R/CfromR/myRLib.c
#include <R.h>

/* sayHello part not shown */

void mySum(double *a, int* n, double *s) {
    int i;
    *s = 0.0;
    for (i = 0; i < *n; i++) {
        *s += a[i];
    }
}

void myMult(double *a, int *n, double *lambda, double *b) {
    int i;
    for (i = 0; i < *n; i++) {
        b[i] = (*lambda)*a[i];
    }
}
